### PR TITLE
Update CMake min to 3.12

### PR DESCRIPTION
**Pull request by powderluv**
STRING JOIN was introduced in 3.12

TEST=builds with cmake 3.12 fails with 3.10.2 (stock in ubuntu)

Latest Ubuntu PPAs are here: https://apt.kitware.com/

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
FYI the code style check failure is spurious. The github CI has been failing for a while. I'm trying to figure out how to disable it until we can get it working properly so it doesn't make people think they broke something

File: iree/base/api_util.h [hunk 1]
```diff
@@ -25,7 +25,6 @@
 namespace iree {
 
 inline iree_status_t ToApiStatus(Status status) {
-  DLOG(ERROR) << status;
```
Comment (benvanik) [inline iree/base/api_util.h]: This log line is intentional - it's the only way to get detailed status messages when using the C API.

Comment (powderluv) [reply iree/base/api_util.h]: How do we get DLOG ? Should it just be LOG(ERROR) ?

Comment (powderluv) [reply iree/base/api_util.h]: I pushed a patch using LOG instead and it seems to work ok. Maybe that is better ?

Comment (benvanik) [reply iree/base/api_util.h]: What you had was fine - it was my mistake for forgetting we didn't have this duplicated in the open source repo yet. Really close to getting a public CI setup so we don't do that again :)

Comment (ScottTodd) [reply iree/base/api_util.h]: We merged the PR upstream with your original change, removing the line. It'll take a few minutes for the upstream change to be mirrored and close out the pull request.

We can make a separate change that exposes DLOG (debug only logging macro) and adds the line back in.

**Pull request closed without merging**